api/controllers/CampaignBotController.js

```diff
@@ -22,433 +17,390 @@ const CMD_REPORTBACK = (process.env.GAMBIT_CMD_REPORTBACK || 'P');
+      const currentSignup = signups[0];
+      this.debug(req, `currentSignup.id:${currentSignup.id}`);
+      const data = {
+        _id: parseInt(currentSignup.id),
+        user: req.user_id,
+        campaign: req.campaign_id,
+        created_at: currentSignup.createdAt,
+      };
+      if (currentSignup.reportback) {
+        data.reportback = parseInt(currentSignup.reportback.id);
+        data.total_quantity_submitted = currentSignup.reportback.quantity;
+      }
 
-    return res.sendStatus(500);
+      return app.locals.db.signups.create(data);
+    })
+    .then(signupDoc => {
+      this.debug(req, `created signupDoc:${signupDoc._id.toString()}`);
+      signup = signupDoc;
+      req.user.campaigns[req.campaign_id] = signupDoc._id;
+      req.user.markModified('campaigns');
+      return req.user.save();
+    })
+    .then(() => {
+      return signup;
+    });
   }
 
-  loadReportbackSubmission(req, res) {
-    this.debug(req, 'loadReportbackSubmission');
-
-    return dbRbSubmissions
-      .findById(this.signup.draft_reportback_submission)
-      .exec()
-      .then(reportbackSubmissionDoc => {
-        this.reportbackSubmission = reportbackSubmissionDoc;
-        const promptUser = (req.query.start || false);
-
-        if (!this.reportbackSubmission.quantity) {
-          return this.collectQuantity(req, res, promptUser);
-        }
-        if (!this.reportbackSubmission.image_url) {
-          return this.collectPhoto(req, res, promptUser);
-        }
-        if (!this.reportbackSubmission.caption) {
-          return this.collectCaption(req, res, promptUser);
-        }
-        if (!this.reportbackSubmission.why_participated) {
-          return this.collectWhyParticipated(req, res, promptUser);
+  /**
+   * Gets User from DS API if exists for given id, else creates new User.
+   * @param {string} id - DS User ID
+   * @return {object} - User model
+   */
+  getUser(id) {
+    return app.locals.clients.northstar.Users
+      .get('id', id)
+      .then(user => {
+        if (!user) {
+          // TODO: Create new User
+          return;
         }
-        // Should have submitted in whyParticipated, but in case of failed_at:
-        logger.warn('no messages sent from chatReportback');
-
-        return this.postReportback(req, res);
+        return app.locals.db.users.create({
+          _id: id,
+          mobile: user.mobile,
+          first_name: user.firstName,
+          email: user.email,
+          phoenix_id: user.drupalID,
+          campaigns: {},   
+        });
       });
   }
 
-  loadSignup(req, res, signupId) {
-    this.debug(req, 'loadSignup');
+  /**
+   * Returns whether incomingMessage should begin a Reportback conversation.
+   * @param {string} incomingMessage
+   * @return {bool}
+   */
+  isCommandReportback(incomingMessage) {
+    const firstWord = helpers.getFirstWord(incomingMessage);
+    return ( firstWord && firstWord.toUpperCase() === CMD_REPORTBACK );
```

Comment:
It'd be better for readability to move this to own line. Also, parentheses are unnecessary.